The CodeQL showed a couple [errors](https://github.com/trilinos/Trilinos/actions/runs/28572546843/job/84713422543?pr=15421#step:9:912) (not directly related to the changes in the pr, but triggered by updating tacho):
```
 In file included from ../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver_double.cpp:11:
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp: In function 'int driver(int, char**)':
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp:401:36: error: 'facto_time' was not declared in this scope
  401 |     std::cout << " Facto Time " << facto_time / (double)nfacts << std::endl;
      |                                    ^~~~~~~~~~
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp:402:36: error: 'solve_time' was not declared in this scope
  402 |     std::cout << " Solve Time " << solve_time / (double)nsolves << std::endl;
      |                                    ^~~~~~~~~~
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp: In instantiation of 'int driver(int, char**) [with value_type = double]':
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver_double.cpp:13:57:   required from here
../packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp:325:16: error: unused variable 'facto_time' [-Werror=unused-variable]
  325 |         double facto_time = timer.seconds();
      |                ^~~~~~~~~~
cc1plus: some warnings being treated as errors
```
@iyamazaki would you have a moment to include an update to address that here?

The ramses-openmp failures should be unrelated, just referencing them [here](https://github.com/trilinos/Trilinos/actions/runs/28572547061/job/84724022748?pr=15421#step:6:5869):
```
 The following tests FAILED:
	109 - TeuchosCore_MemoryManagementExt_UnitTests_MPI_1 (Not Run) Teuchos
	678 - TpetraCore_UnpackLongRows_MPI_4 (Not Run)         Tpetra
	960 - Belos_Kokkos_GCRODRComplexEx_MPI_4 (Failed)       Belos
	2462 - TrilinosCouplings_Example_Poisson_STK_MPI_1 (Timeout) TrilinosCouplings
```

--- a/packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp
+++ b/packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp
@@ -409,6 +409,9 @@ template <typename value_type> int driver(int argc, char *argv[]) {
               << ",\t avg. over (" << niters << " x " << nfacts << ") calls" << std::endl;
     std::cout << " Solve Time " << solve_time / (double)(niters*nsolves)
               << ",\t avg. over (" << niters << " x " << nsolves << ") calls" << std::endl;
+    if (!success) {
+      std::cerr << "\n Error: Some of the residual norms were too large\n\n";
+    }
 #endif
     if (verbose) {
       std::cout << std::endl;

--- a/packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp
+++ b/packages/shylu/shylu_node/tacho/example/Tacho_ExampleDriver.hpp
@@ -301,15 +301,22 @@ template <typename value_type> int driver(int argc, char *argv[]) {
 
     bool success = true;
     mag_type tol = sqrt(arith_traits::epsilon()); // loose accuracy tol..
+#ifndef TACHO_HAVE_TEUCHOS
+    double facto_time = 0.0;
+    double solve_time = 0.0;
+#endif
     for (int iter = 0; iter < niters; iter++) {
       try {
         /// symbolic structure can be reused
         std::cout << "\n === Iteration " << iter << " === " << "\n";
 
         /// factorize
+        double warmup_facto_time = 0.0;
         if (!no_warmup) {
           // warm-up
+          timer.reset();
           solver.factorize(values_on_device);
+          warmup_facto_time = timer.seconds();
         }
 
 #ifdef TACHO_HAVE_TEUCHOS
@@ -322,20 +329,20 @@ template <typename value_type> int driver(int argc, char *argv[]) {
         for (int i = 0; i < nfacts; ++i) {
           solver.factorize(values_on_device);
         }
-        double facto_time = timer.seconds();
+        facto_time += timer.seconds();
 #endif
 
         /// solve
         bool pass = true;
-        double solve_time = 0.0;
         mag_type shift = solver.currentShift();
         if (!no_warmup) {
           // warm-up
           timer.reset();
           solver.solve(x, b, t);
-          solve_time = timer.seconds();
+          double warmup_solve_time = timer.seconds();
           const double res = solver.computeRelativeResidual(values_on_device, x, b, shift);
-          std::cout << "TachoSolver (warm-up): residual = " << res << " time " << solve_time;
+          std::cout << "TachoSolver (warm-up): residual = " << res
+                    << " time " << warmup_facto_time << " + " << warmup_solve_time;
           if (shiftDiag) std::cout << " using shift = " << shift;
           std::cout << "\n";
           if (res > tol) pass = false;
@@ -398,8 +405,10 @@ template <typename value_type> int driver(int argc, char *argv[]) {
     std::cout << std::endl;
 #else
     std::cout << " Initi Time " << initi_time << std::endl;
-    std::cout << " Facto Time " << facto_time / (double)nfacts << std::endl;
-    std::cout << " Solve Time " << solve_time / (double)nsolves << std::endl;
+    std::cout << " Facto Time " << facto_time / (double)(niters*nfacts)
+              << ",\t avg. over (" << niters << " x " << nfacts << ") calls" << std::endl;
+    std::cout << " Solve Time " << solve_time / (double)(niters*nsolves)
+              << ",\t avg. over (" << niters << " x " << nsolves << ") calls" << std::endl;
 #endif
     if (verbose) {
       std::cout << std::endl;

